src/parsers/transactions.ts

```diff
@@ -0,0 +1,535 @@
+    txn => !!txn.hash && !txn.pending
+  );
+  const lastTxHash = lastSuccessfulTxn?.hash;
+  if (lastTxHash) {
+    const lastTxnHashIndex = findIndex(transactionData, txn =>
+      lastTxHash.startsWith(txn.hash)
+    );
+    if (lastTxnHashIndex > -1) {
+      return slice(transactionData, 0, lastTxnHashIndex + LAST_TXN_HASH_BUFFER);
+    }
+  }
+  return transactionData;
+};
+
+export const parseTransactions = (
+  transactionData: ZerionTransaction[],
+  accountAddress: string,
+  nativeCurrency: string,
+  existingTransactions: RainbowTransaction[],
+  purchaseTransactions: RainbowTransaction[],
+  network: string,
+  appended = false
+) => {
+  const purchaseTransactionHashes = map(purchaseTransactions, txn =>
+    ethereumUtils.getHash(txn)
+  );
+  const data = appended
+    ? transactionData
+    : dataFromLastTxHash(transactionData, existingTransactions);
+
+  const parsedNewTransactions = flatten(
+    data.map(txn =>
+      parseTransaction(
+        txn,
+        accountAddress,
+        nativeCurrency,
+        purchaseTransactionHashes,
+        network
+      )
+    )
+  );
+
+  const [pendingTransactions, remainingTransactions] = partition(
+    existingTransactions,
+    txn => txn.pending
+  );
+
+  const updatedPendingTransactions = dedupePendingTransactions(
+    accountAddress,
+    pendingTransactions,
+    parsedNewTransactions
+  );
+
+  const updatedResults = concat(
+    updatedPendingTransactions,
+    parsedNewTransactions,
+    remainingTransactions
+  );
+
+  const potentialNftTransaction = appended
+    ? find(parsedNewTransactions, txn => {
+        return (
+          !txn.protocol &&
+          (txn.type === 'send' || txn.type === 'receive') &&
+          txn.symbol !== 'ETH'
+        );
+      })
+    : null;
+
+  const dedupedResults = uniqBy(updatedResults, txn => txn.hash);
+
+  const orderedDedupedResults = orderBy(
+    dedupedResults,
+    ['minedAt', 'nonce'],
+    ['desc', 'desc']
+  );
+
+  return {
+    parsedTransactions: orderedDedupedResults,
+    potentialNftTransaction,
+  };
+};
+
+const transformTradeRefund = (
+  internalTransactions: ZerionTransactionChange[]
+) => {
+  const [txnsOut, txnsIn] = partition(
+    internalTransactions,
+    txn => txn?.direction === TransactionDirection.out
+  );
+  const isSuccessfulSwap =
+    txnsOut.length === 1 && (txnsIn.length === 1 || txnsIn.length === 2);
+  if (!isSuccessfulSwap) return internalTransactions;
+
+  const txnOut = txnsOut[0];
+  const txnIn = find(
+    txnsIn,
+    txn => txn?.asset?.asset_code !== txnOut?.asset?.asset_code
+  );
+  const refund = find(
+    txnsIn,
+    txn => txn?.asset?.asset_code === txnOut?.asset?.asset_code
+  );
+  let updatedOut = txnOut;
+  if (refund?.value && txnOut?.value) {
+    updatedOut = {
+      ...txnOut,
+      value: txnOut.value - refund.value,
+    };
+  }
+  return compact([updatedOut, txnIn]);
+};
+
+const overrideFailedCompound = (
+  txn: ZerionTransaction,
+  network: string
```

Comment:
I swear we have a type for networks (or an Enum)

Reply:
yes we do :D was going to do that after the txns one as part of a separate PR as it touches a bunch of files